documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,457 @@
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process to collect a process dump"
+        required: true
+        type: "integer"
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultgcdump"
+      produces:
+      - "application/octet-stream"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process to collect a process dump"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /cpuprofile:
+    get:
+      tags:
+      - "traces"
+      operationId: "getdefaultcputrace"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "duration"
+        in: "query"
+        description: "Duration of the CPU trace"
+        required: false
+        type: "integer"
+        default: "30"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /cpuprofile/{uid}:
+    get:
+      tags:
+      - "traces"
+      operationId: "getcputrace"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process to collect a process dump"
```

Comment:
Some copypaste error here i think

Reply:
Ahh. I'll change it to just process id. The rest of the text belongs in the description on the operation